### DISCUSSION TITLE
Error: Converting from type System.String to abap type DATE is not supported

### DISCUSSION THREAD

**Discussion by victorfrco** [dbosoft/YaNco#228] [Q&A] [answered]
Hey guys, I have the following implementation:

![image](https://user-images.githubusercontent.com/19866419/236881477-2d06c48d-3c1c-4027-9764-ed0fc3fc79bf.png)

But I'm receiving the following message:
`[ERR] SAP30Connection::CreateAfterClientEventAsync(): Error: Converting from type System.String to abap type DATE is not supported.`

what am I missing?

Thanks in advance.

**Comment by fw2568** [comment, marked as answer]
Hi,

> what am I missing?

A string is not a date :grin:

Just pass the DateTime, YanCo will automatically convert to SAP format.

**Comment by NDayakarChowdary** [comment]
Hey Hi ,Good Day!!
Can we Know what are the available Data Types YanCo will automatically convert to SAP format?

**Reply by fw2568** [reply]
Good point, the type conversation is not currently listed in the readme. I have added an issue to add this #269.

Conversion is configurable, the default converters are implemented here: https://github.com/dbosoft/YaNco/tree/main/src/YaNco.Core/TypeMapping
